File: ruby/test/media_models_test.rb
# frozen_string_literal: true

require_relative 'test_helper'

class MediaModelsTest < Minitest::Test
  def test_lists_media_models_with_media_type_filter
    transport = FakeTransport.new
    transport.enqueue_response('data' => [])
    resource = Buble::MediaModelsResource.new(transport)

    resource.list(media_type: 'video')

    assert_equal 'GET', transport.requests[0][:method]
    assert_equal '/api/v1/media_models', transport.requests[0][:path]
    assert_equal({ media_type: 'video' }, transport.requests[0][:query])
  end
end

File: elixir/test/buble/media_models_test.exs
defmodule Buble.MediaModelsTest do
  use ExUnit.Case, async: true

  test "lists media models with query filters" do
    {:ok, client} = Buble.Client.new(api_key: "sk-test", transport: Buble.TestTransport)

    Process.put(:buble_test_handler, fn _client, method, path, opts ->
      assert method == :get
      assert path == "/api/v1/media_models"
      assert Keyword.fetch!(opts, :query) == %{"media_type" => "image"}
      {:ok, %{"data" => []}}
    end)

    assert {:ok, %{"data" => []}} = Buble.MediaModels.list(client, media_type: "image")
  end
end
